zerver/tests/test_embedded_bot_system.py

```diff
@@ -101,12 +108,13 @@ def test_message_embedded_bot_with_invalid_service(self) -> None:
         service_profile.name = "invalid"
         service_profile.save()
         with self.assertLogs(level="ERROR") as m:
-            self.send_stream_message(
-                user_profile,
-                "Denmark",
-                content=f"@**{bot_profile.full_name}** foo",
-                topic_name="bar",
-            )
+            with self.captureOnCommitCallbacks(execute=True):
+                self.send_stream_message(
+                    user_profile,
+                    "Denmark",
+                    content=f"@**{bot_profile.full_name}** foo",
+                    topic_name="bar",
+                )
```

Comment:
Same as above, about this being `_send_stream_message` (or whatever we end up calling that function, if it does end up being distinct from `send_stream_message`)

Reply:
It is just that `_send_stream_message` is defined for `TestEmbeddedBotMessaging` class only (did so since we only define function inside classes in most of the tests).

Writing a separate function in `test_classes.py` as mentioned [here](https://github.com/zulip/zulip/pull/22737/commits/1a33f1ab6dff396fcbf7ccfafe293af508db922c#r1036310300) would be better.